### PR TITLE
WIP: Liveview

### PR DESCRIPTION
**Pull request by gemantzu**
This is why I hate frontend... Live loading works, but what now needs to be done is to adapt the search function to preload jobs / industries as `all/1` does.

### CODE REVIEW

**Comment by gemantzu** [comment]
@doomspork take a look, liveview works except hiring. Plus we could use @burden 's or @tajchumber 's help on the form looks.

File: lib/companies/jobs.ex [hunk 1]
```diff
@@ -102,4 +102,20 @@ defmodule Companies.Jobs do
   def change(%Job{} = job) do
     Job.changeset(job, %{})
   end
+
+  def search(filters) do
+    filters
+    |> Enum.reduce(Job, fn
+      {_, nil}, query ->
```
Comment (doomspork) [inline lib/companies/jobs.ex]: Let's make this a named function instead. I've never been a big fan of anonymous functions w/ many heads.

How about something like this? I _think_ we can also reduce this to just 2 matches:

```elixir
defp query_predicates({"text", text}, query), 
  do: from c in query, where: ilike(c.title, ^"%#{text}%")
defp query_predicates(_, query), 
  do: query
```

Comment (gemantzu) [reply lib/companies/jobs.ex]: Sure, I will do the change. Still adding stuff btw.

**Comment by gemantzu** [comment]
I am at my limit with JS here, can't really tell why it runs like that. Thoughts @doomspork  / @tajchumber / @SophieDeBenedetto ?

**Comment by gemantzu** [comment]
Closed in favor of #534 (non LiveView)

**Pull request closed without merging**